ryan_savill/03_background_subtraction/readme.md

```diff
@@ -0,0 +1,281 @@
+
+
+```python
+from skimage import filters
+
+# the smaller gaussian is just there to remove 
+# noise so it can be just a few pixels
+small_gauss = filters.gaussian(tribolium, sigma = 1)
+
+# the larger gaussian has to be bigger than our largest
+# object, which we can measure or make an educated guess
+large_gauss = filters.gaussian(tribolium, sigma = 100)
+
+# now we subtract the large gaussian from the small one
+dog_tribolium = small_gauss - large_gauss
+
+# visualising the results
+fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize =(20,20))
+ax1.imshow(small_gauss, cmap = 'gray')
+ax1.set_title('Small Gaussian')
+ax2.imshow(large_gauss, cmap = 'gray')
+ax2.set_title('Large Gaussian')
+ax3.imshow(dog_tribolium, cmap = 'gray')
+ax3.set_title('DoG Image')
+```
+
+
+    
+![png](output_19_1.png)
+    
+
+
+We can see that it is not quite as effective as the tophat-filter but nonetheless, it has managed to remove some of the background in the image. Making this filter ourselves is a good exercise in coding and understanding what these filters do but when it gets to complicated filters reinventing the wheel isn't practical and you will most likely find the filter you are looking for either in [scikit image](https://scikit-image.org/), in [scipy](https://docs.scipy.org/doc/) or just google (this is what I often do). Because python is a relatively simple language to learn it has many users who have most likely asked the same question as you have countless times! For the difference of gaussian filter scikit image has this function, which we can use here instead of doing the processing ourselves:
+
+
+```python
+# setting the same sigmas as above
+sk_dog_tribolium = filters.difference_of_gaussians(tribolium, 1, 100)
+
+# visualising the results
+# visualising the results
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize =(20,20))
+ax1.imshow(tribolium, cmap = 'gray')
+ax1.set_title('Original')
+ax2.imshow(sk_dog_tribolium, cmap = 'gray')
+ax2.set_title('DoG Image')
+```
+
+
+    
+![png](output_21_1.png)
+    
+
+
+Difference of gaussian is not only helpful as a background filter but it can also be used in spot- or blob-detection, which we will look at in the [next post]! Until then happy coding!
```

Comment:
```suggestion
Difference of Gaussian is not only helpful as a background subtraction filter, it can also be used in spot- or blob-detection, which we will look at in the [next post]! Until then happy coding!
```